packages/devtools_app/lib/src/analytics/analytics.dart

```diff
@@ -474,18 +475,53 @@ void setupAndGaScreen(String screenName) async {
   }
 }
 
-Future<void> setupDimensions() async {
-  if (serviceManager.connectedApp != null &&
-      isGtagsEnabled() &&
-      !isDimensionsComputed &&
-      !_computing) {
-    _computing = true;
-    // While spinning up DevTools first time wait until dimensions data is
-    // available before first GA event sent.
-    await computeUserApplicationCustomGTagData();
+void setupDimensions() {
+  if (isGtagsEnabled() && !_analyticsComputed && !_computingDimensions) {
+    _computingDimensions = true;
     computeDevToolsCustomGTagsData();
     computeDevToolsQueryParams();
     computeFlutterClientId();
-    dimensionsComputed();
+    _analyticsComputed = true;
+  }
+}
+
+Future<void> setupUserApplicationDimensions() async {
+  if (serviceManager.connectedApp != null &&
+      !_userApplicationDimensionsComputed &&
+      !_computingUserApplicationDimensions) {
+    _computingUserApplicationDimensions = true;
+    await computeUserApplicationCustomGTagData();
+    _userApplicationDimensionsComputed = true;
   }
 }
+
+Map<String, dynamic> generateSurveyQueryParameters() {
+  const clientIdKey = 'ClientId';
+  const ideKey = 'IDE';
+  const fromKey = 'From';
+  const internalKey = 'Internal';
+
+  // TODO(https://github.com/flutter/devtools/issues/2475): fix url structure
+  // Parsing the url via Uri.parse returns an incorrect value for fragment.
+  // Grab the fragment value manually. The url will be of the form
+  // http://127.0.0.1:9100/#/timeline?ide=IntelliJ-IDEA&uri=..., and we want the
+  // part equal to '/timeline'.
+  final url = window.location.toString();
+  const fromValuePrefix = '#/';
+  final startIndex = url.indexOf(fromValuePrefix);
+  final endIndex = url.indexOf('?');
+  final fromValue = url.substring(
```

Comment:
Is this the profile page e.g., timeline, memory, etc ?  If so, maybe a better variable name profilePage and profilePagePrefix?

Reply:
changed to `fromPage`